hydroflow_lang/src/graph/hydroflow_graph.rs

```diff
@@ -552,19 +556,24 @@ impl HydroflowGraph {
     pub fn insert_subgraph(
         &mut self,
         node_ids: Vec<GraphNodeId>,
+        laziness: bool,
     ) -> Result<GraphSubgraphId, (GraphNodeId, GraphSubgraphId)> {
         // Check none are already in subgraphs
         for &node_id in node_ids.iter() {
             if let Some(&old_sg_id) = self.node_subgraph.get(node_id) {
                 return Err((node_id, old_sg_id));
             }
         }
-        Ok(self.subgraph_nodes.insert_with_key(|sg_id| {
+        let subgraph_id = self.subgraph_nodes.insert_with_key(|sg_id| {
             for &node_id in node_ids.iter() {
                 self.node_subgraph.insert(node_id, sg_id);
             }
             node_ids
-        }))
+        });
+
+        self.subgraph_laziness.insert(subgraph_id, laziness);
```

Comment:
Lets make subgraph laziness a separate function, `set_subgraph_laziness` so it is less prominent and matches how stratums are handled. (getter optional)

```rust
    /// Gets the stratum nubmer of the subgraph.
    pub fn subgraph_stratum(&self, sg_id: GraphSubgraphId) -> Option<usize> {
        self.subgraph_stratum.get(sg_id).copied()
    }

    /// Set subgraph's stratum number, returning the old value if exists.
    pub fn set_subgraph_stratum(
        &mut self,
        sg_id: GraphSubgraphId,
        stratum: usize,
    ) -> Option<usize> {
        self.subgraph_stratum.insert(sg_id, stratum)
    }
```